hawkmoth/parser.py

```diff
@@ -293,6 +293,34 @@ def _get_access_specifier(cursor):
 
     return name_map.get(cursor.access_specifier, None)
 
+def _specifiers_fixup(cursor, basetype):
+    """Fix the type for C++ specifiers.
+
+    Note the ``basetype`` is not necessarily ``cursor.type``. When dealing with
+    pointers or arrays, we need to get to the base type as in
+    :py:func:`__type_fixup`.
+
+    Returns:
+        List of C++ specifiers for the cursor.
+    """
+    type_elem = []
+
+    if cursor.kind in [CursorKind.VAR_DECL,
+                       CursorKind.FIELD_DECL,
+                       CursorKind.PARM_DECL]:
+        tokens = [t.spelling for t in cursor.get_tokens()]
+
+        if 'mutable' in tokens:
+            type_elem.append('mutable')
+
+        if 'constexpr' in tokens:
+            type_elem.append('constexpr')
+            type_elem.append(basetype.spelling[len('const '):])
```

Comment:
Yikes that feels magic.

Reply:
I actually wondered for a second or two whether that might be the case, but ultimately decided it was quite obvious. Good having someone checking my assumptions :)

So `constexpr` gets translated by Clang to `const` as part of the type spelling, which it is as the 1st implies the 2nd. However we want `constexpr` instead since it has a broader meaning than just `const`. We could leave it as `constexpr const`, but I think it looks a bit ugly and redundant. I'll add a comment to that effect.